src/Server/Event/Listener/CollectionListener.php

```diff
@@ -0,0 +1,107 @@
+    {
+        $this->listeners[] = $events->attach(
+            DoctrineResourceEvent::EVENT_UPDATE_PRE,
+            array( $this, 'handleCollections' )
+        );
+
+        $this->listeners[] = $events->attach(
+            DoctrineResourceEvent::EVENT_CREATE_PRE,
+            array( $this, 'handleCollections' )
+        );
+    }
+
+    public function handleCollections(DoctrineResourceEvent $event)
+    {
+        $objectManager = $event->getObjectManager();
+        $entity        = $event->getEntity();
+        $originalData  = (array) $event->getData();
+
+        $metadata = $objectManager->getClassMetadata(get_class($entity));
+
+        $associations = $metadata->getAssociationNames();
+
+        foreach ($associations as $association) {
+            // Skip handling associations that arent in the data
+            if ($metadata->isCollectionValuedAssociation($association)) {
+                if (array_key_exists($association, $originalData)
+                    && !empty( $originalData[$association] )
+                    && ( is_array($originalData[$association])
+                         || $originalData[$association] instanceof \Traversable )
+                ) {
+                    foreach ($originalData[$association] as &$subEntityData) {
+                        $target          = $metadata->getAssociationTargetClass($association);
+                        $identifierNames = $metadata->getIdentifierFieldNames($target);
+                        if (empty( $identifierNames )) {
+                            // TODO Investigate what we should do here, throw exception? for now handle downstream
+                            continue;
+                        }
+
+                        $identifierValues = [ ];
+                        foreach ($identifierNames as $identifierName) {
+                            if (!isset( $subEntityData[$identifierName] ) || empty( $subEntityData[$identifierName] )) {
+                                continue;
+                            }
+                            $identifierValues[$identifierName] = $subEntityData[$identifierName];
+                        }
+
+                        $subEntity = false;
+                        if (count($identifierValues) === count($identifierNames)) {
+                            $subEntity = $objectManager->find($target, $identifierValues);
+                        }
+
+                        if (!$subEntity) {
+                            $subEntity = new $target;
+                        }
+
+                        // TODO This could cause a catastrophe as it wouldnt have the appropriate strategies/listeners
+                        // probably should fetch from "hydrator-manager" if possible
+                        $hydrator = new DoctrineObject($objectManager);
+                        $hydrator->hydrate($subEntityData, $subEntity);
+                        $objectManager->persist($subEntity);
+                        $objectManager->flush();
```

Comment:
Events in #182 don't flush until after the POST event.  Do you need to flush here or can we wrap the whole UOW in the DoctrineResource?